StarVMC/StarGeometry doesn't compile on main branch

### Steps to reproduce

On RACF run:
```bash
git clone https://github.com/star-bnl/star-sw.git
cd star-sw
git checkout f7935bbbca1c9e9c69931bb5167ca9767be72807
env -i \
  PATH=$PWD/mgr:$PATH \
  QTDIR=/usr/lib64/qt-3.3 QTINC=/usr/lib64/qt-3.3/include \
  QTLIB=/usr/lib64/qt-3.3/lib CERN_LEVEL=pro \
  CERN=/cern CERN_ROOT=/cern/pro \
  ROOTSYS=/afs/rhic.bnl.gov/star/ROOT/5.34.38/.sl73_gcc485/rootdeb \
  ROOT=/afs/rhic.bnl.gov/star/ROOT \
  ROOT_LEVEL=5.34.38 \
  STAR_SYS=x8664_sl7 \
  STAR_HOST_SYS=sl73_gcc485 \
  OPTSTAR=/opt/star/sl73_gcc485 \
  XOPTSTAR=/afs/rhic.bnl.gov/opt/star/sl73_gcc485 \
  STAR_ROOT=/afs/rhic.bnl.gov/star \
  STAR=/nowhere \
  sh -c '. $ROOTSYS/bin/thisroot.sh; exec cons %StShadowMaker'
```

Observe:
```
Install StarVMC/StarGeometry/BtofConfig.cxx as .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.cxx
g++ -m32 -fPIC -pipe -Wall -Woverloaded-virtual -std=c++0x -Wno-long-long -pthread -g -Dsl73_gcc485 -D__ROOT__ -DNEW_DAQ_READER -I. -IStRoot -I.sl73_gcc485/include -I/afs/rhic.bnl.gov/star/ROOT/5.34.38/.sl73_gcc485/rootdeb/include -c .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.cxx -o .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.o
In file included from .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.cxx:1:0:
./StarVMC/StarGeometry/BtofConfig.h:3:45: fatal error: StarVMC/StarGeometry/BtofGeo1.h: No such file or directory # include "StarVMC/StarGeometry/BtofGeo1.h"                                              ^
compilation terminated.
cons: *** [.sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.o] Error 1
cons: errors constructing .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.o
```

### Expected result

Everything compiles without errors

cc @klendathu2k

--- a/StarVMC/Geometry/StarGeoLinkDef.h
+++ b/StarVMC/Geometry/StarGeoLinkDef.h
@@ -1,0 +1,8 @@
+#ifdef __CINT__
+#pragma link off all globals;
+#pragma link off all classes;
+#pragma link off all functions;
+
+#pragma link C++ class StarGeometry;
+#pragma link C++ class Geometry;
+#endif

--- a/docker/Dockerfile.root5
+++ b/docker/Dockerfile.root5
@@ -122,7 +122,6 @@ diff --git a/asps/rexe/Conscript b/asps/rexe/Conscript \n\
 
 RUN source /etc/profile \
  && cd /star-sw \
- && cons +StarVMC/Geometry \
  && cons %StRoot/StShadowMaker \
  && find /star-sw/.$STAR_HOST_SYS -name *.o -exec rm '{}' \;
 

--- a/docker/Dockerfile.root6
+++ b/docker/Dockerfile.root6
@@ -125,7 +125,6 @@ RUN source /etc/profile \
  # Override default Vc version
  && export Vc_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/vc_-0.7.4-gqbhzu2x5u2dbe5tafxnl36xj5qbwov4 \
  && cd /star-sw \
- && cons +StarVMC/Geometry \
  && cons %StRoot/StShadowMaker \
  && find /star-sw/.$STAR_HOST_SYS -name *.o -exec rm '{}' \;
 

--- a/mgr/Conscript-standard
+++ b/mgr/Conscript-standard
@@ -376,33 +376,22 @@ if ( $pkg !~ /^sim$/ && $pkg !~ /^gen$/ ) {
 			$env->{ENV}->{AGML_WARNINGS} = 1==1; # $param::debug;
 
 			if ( $xml =~ m/Compat/ ){
-			    Depends $env ["#".$agE], ("#$xml", "#$agml_cmd");
-			    Command $env ["#".$agE], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=Mortran > %>);
+			    Depends $env ["$OBJ/$agE"], ("#$xml", "#$agml_cmd");
+			    Command $env ["$OBJ/$agE"], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=Mortran > %>);
 
 			} else {
-			    Depends $env ["#".$agX, "#".$agH, "#".$agE], ("#$xml", "#$agml_cmd");
+			    Depends $env ["$OBJ/$agX", "$OBJ/$agH", "$OBJ/$agE"], ("#$xml", "#$agml_cmd");
 			    # Depends $env --> some rule to build extra dict on the fly
 
-			    Command $env ["#".$agX], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=AgROOT --path=%>:d);
-			    Command $env ["#".$agE], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=Mortran > %>);
+			    Command $env ["$OBJ/$agX", "$OBJ/$agH"], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=AgROOT --path=%>:d);
+			    Command $env ["$OBJ/$agE"], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=Mortran > %>);
 
-			    # Install local file (#) as global
-			    # The .h file can be installed as-is
-			    $trgt = $OBJ."/".$agH;
-			    InstallAs  $env $trgt, "#".$agH;
-
-			    # The recently generated .cxx will create an extra source dependency
-			    $trgt = $OBJ."/".$agX;
-			    InstallAs  $env $trgt, "#".$agX;
+			    $trgt = $INCLUDE."/".$agH;
+			    InstallAs  $env $trgt, "$OBJ/$agH";
 
 			    # Add source dependency but only cxx
 			    #>>push(@srcL, $trgt);
 			}
-
-			# Install the "#" i.e. local file as global file - ageF created in both cases
-			# BEWARE that the agE (.age) file is created in xgeometry
-			$trgt = $OBJ."/".$agE;
-			InstallAs  $env $trgt, "#".$agE;
 
 			# Create source dependencies based on installed age, not the original
 			#>>push(@srcL, $trgt );

--- a/mgr/Conscript-standard
+++ b/mgr/Conscript-standard
@@ -304,6 +304,12 @@ if ( $pkg !~ /^sim$/ && $pkg !~ /^gen$/ ) {
 	}
 	#print "\tAfter  ".($#src)." ".($#inc)."\n";
 
+        if ($Dir =~ m#StarVMC/StarGeometry#) {
+            Command $env ["$OBJ/StarVMC/StarGeometry/StarGeo_Cint.cxx", "$OBJ/StarVMC/StarGeometry/StarGeo_Cint.h"],
+                         ("$OBJ/StarVMC/StarGeometry/StarGeo.h", "#StarVMC/Geometry/StarGeoLinkDef.h"), qq(rootcint -f %> -c %1 %2);
+            push(@srcL, "$OBJ/StarVMC/StarGeometry/StarGeo_Cint.cxx");
+        }
+
 	#print "-=-=-=-=- LALALA -=-=-=-\n";
 	# 2018/01 issue after updating to AgML 2.0
 	# TODO problem with optimization in xgeometry and StarGeometry
@@ -1118,7 +1124,7 @@ if ( $pkg !~ /^sim$/ && $pkg !~ /^gen$/ ) {
 	}
     }
 
-    if ( $#Defs > -1 ){  # && ! $OnlinePackage ) {
+    if ( $#Defs > -1 && $pkg !~ m/Geometry/ ) {
 	my @CintFiles = ( $PKG . "_Cint.cxx", $PKG . "_Cint.h", "LinkDef.h" );
 
         if ($env->{ENV}->{ROOT_VERSION_MAJOR} >= 6) {

--- a/mgr/Conscript-standard
+++ b/mgr/Conscript-standard
@@ -288,17 +288,13 @@ if ( $pkg !~ /^sim$/ && $pkg !~ /^gen$/ ) {
 	    $trgt = "";
 	    $hfl  = "";
 	    if ( $Dir =~ m/xgeometry/ ){
-		$fcheck = $trgt  = $OBJ."/".$Dir."/".$agM.".age";
-		$fcheck =~ s/\#//; $trgt = "" if ( -e $fcheck);
+		$trgt = "$OBJ/$Dir/$agM.age";
 	    } else {
 		# Compat contains redundant defs only needed for the age
 		# .h also needed for Dictionary building
 		if ( $xml !~ m/Compat/ ){
-		    $fcheck = $trgt  = $OBJ."/".$Dir."/".$agM.".cxx";
-		    $fcheck =~ s/\#//; $trgt = "" if ( -e $fcheck);
-
-		    $fcheck = $hfl   = $OBJ."/".$Dir."/".$agM.".h";
-		    $fcheck =~ s/\#//; $hfl = "" if ( -e $fcheck);
+		    $trgt = "$OBJ/$Dir/$agM.cxx";
+		    $hfl  = "$OBJ/$Dir/$agM.h";
 		}
 	    }
 	    # push but only if not present - if present, cons will find

